### PR TITLE
WIP: Qemu q35 v0.1.0 rc5

### PR DESCRIPTION
**Pull request by pietrushnic**
@macpijan I'm confused with the rebase of secure boot keywords. This PR consists of the changes I created on top of the previous version of the code. With those changes, I was able to enable a secure boot in QEMU by resetting keys successfully.

I thought I could address self-test failures mentioned here: https://github.com/Dasharo/open-source-firmware-validation/pull/95#issuecomment-1803857495

But I have no idea how lib keywords are related to those used in self-test. From what I see, those are different keywords. So, what is the purpose of self-test?

### CODE REVIEW

**Comment by pietrushnic** [comment]
The procedure worked well on https://github.com/Dasharo/open-source-firmware-validation/tree/qemu_q35_v0.1.0-rc4 branch

**Comment by macpijan** [comment]
> But I have no idea how lib keywords are related to those used in self-test

They are keywords lib contains keyowrds, self-tests contains **tests**

File: keywords.robot [hunk 1]
```diff
@@ -1886,7 +1886,7 @@ Remove Entry From List
 
 Get Secure Boot Configuration Submenu Construction
```
Comment (macpijan) [inline keywords.robot:1887]: I overlooked this kwd. This should be gone from the keywords.robot.

File: lib/secure-boot-lib.robot [hunk 2]
```diff
@@ -236,7 +236,64 @@ Check Secure Boot In Windows
     ...    Should Contain    ${out}    True
     RETURN    ${sb_status}
 
-Check Gf Attempt Secure Boot Can Be Selected
+Enable Secure Boot
```
Comment (macpijan) [inline lib/secure-boot-lib.robot:239]: I'm not sure if we need this kwd. Wy try to favor smaller steps in keywords, so you can get a good idea on the test scenario just by reading the test case (which will be generated as a documentation), so it can also be reproduced manually.   It can be implemented in just 3 lines with this MR: https://github.com/Dasharo/open-source-firmware-validation/pull/95

If we will use it often, then let's add it, but as below.
```
    ${sb_menu}=    Enter Secure Boot Menu And Return Construction
    Set Option State    ${sb_menu}    Enable Secure Boot    ${TRUE}
    Save Changes And Reset    2
```

File: lib/secure-boot-lib.robot [hunk 3]
```diff
@@ -236,7 +236,64 @@ Check Secure Boot In Windows
...
+    ${device_mgr_menu}=    Enter Submenu From Snapshot And Return Construction
+    ...    ${setup_menu}
+    ...    Device Manager
+    ${secure_boot_menu}=    Enter Submenu From Snapshot And Return Construction
+    ...    ${device_mgr_menu}
+    ...    Secure Boot Configuration
+    Select Enable Secure Boot Option    ${secure_boot_menu}
+    Save Changes And Reset    2
+
+Disable Secure Boot
```
Comment (macpijan) [inline lib/secure-boot-lib.robot:250]: As above

File: lib/secure-boot-lib.robot (hunk 2 above)
Comment (macpijan) [reply lib/secure-boot-lib.robot:239]: Or maybe we do need them, I'm not sure at this point...
https://github.com/Dasharo/open-source-firmware-validation/pull/95/files#diff-50de80694206570830c0d9e0dd8bc964352cf44db51a4c3eab6368dab4dea03fR123

File: lib/secure-boot-lib.robot [hunk 4]
```diff
@@ -236,7 +236,64 @@ Check Secure Boot In Windows
...
+
+Enter Custom Secure Boot Options
+    [Documentation]    Sets the Secure Boot Mode to Custom Mode from
+    ...    the Secure Boot Configuration menu and enters
+    ...    the Custom Secure Boot Options menu
+    ...
+    ${sb_menu}=    Reenter Menu And Return Construction
+    ${is_standardmode}=    Run Keyword And Return Status
+    ...    Should Contain Match    ${sb_menu}    *Standard*
+    IF    ${is_standardmode}
```
Comment (macpijan) [inline lib/secure-boot-lib.robot:269]: If possible, we should avoid hard-coding menu positions and use menu construction lists to select entries instead. Kwd for entering advanced menu is already here: https://github.com/Dasharo/open-source-firmware-validation/pull/95/files#diff-b06ff90d8f021dfa2e18d217f3d4c1e6861d6f671893cf292b6718b80227921aR49

While just selecting certain option from the lit is as as simple as: ` Set Option State    ${sb_menu}    Secure Boot Mode    Custom Mode`

This kwd should take care of returning selectable entries only: https://github.com/Dasharo/open-source-firmware-validation/pull/95/files#diff-b06ff90d8f021dfa2e18d217f3d4c1e6861d6f671893cf292b6718b80227921aR9 
With the new many layout it is a bit easier to do.

File: lib/secure-boot-lib.robot (hunk 2 above)
Comment (pietrushnic) [reply lib/secure-boot-lib.robot:239]: @macpijan I feel we wasted some energy on this coordination because you seem to face the same issues I faced and tried to resolve. Of course, my solution may not be up to the expected quality, but that should be the point of your review. I'm unsure how to resolve those experiments and ensure we inform each other about vision, plans, and achievements so the other side will not repeat the same mistakes and will not be affected by releases, API changes, or different visions of implementation.

Please note `Select Enable Secure Boot Option` use ` Check If Enable Secure Boot Can Be Selected ` to handle precisely your concern from the linked TODO comment. Is it a good design? That is the question.

File: lib/secure-boot-lib.robot (hunk 4 above)
Comment (pietrushnic) [reply lib/secure-boot-lib.robot:269]: @macpijan that is very useful I was not aware that there is code that handle that. I will try to use that code.

File: keywords.robot (hunk 1 above)
Comment (pietrushnic) [reply keywords.robot:1887]: I will remove it.

Comment (pietrushnic) [reply keywords.robot:1887]: Ok I see you already did that https://github.com/Dasharo/open-source-firmware-validation/commit/0567475d5abec3b7f8a769f2d122054ca59a45db

File: lib/secure-boot-lib.robot (hunk 2 above)
Comment (macpijan) [reply lib/secure-boot-lib.robot:239]: @pietrushnic Please just use my MR: https://github.com/Dasharo/open-source-firmware-validation/pull/95#issuecomment-1805513832

File: lib/secure-boot-lib.robot (hunk 4 above)
Comment (macpijan) [reply lib/secure-boot-lib.robot:269]: That was in my MR (submitted before this one), which means you have probably not looked at that before submitting this. I specifically drafted a plan here: https://github.com/Dasharo/open-source-firmware-validation/pull/95#issue-1984603821 and pointed to: `Make sure to review/rework all kwds in secure boot lib - see this comment: https://github.com/Dasharo/open-source-firmware-validation/pull/95/files#diff-b06ff90d8f021dfa2e18d217f3d4c1e6861d6f671893cf292b6718b80227921aR107 - put useful/tested kwds above that comment, and eventually remove it` to indicate which keywords I see as of a reasonable quality, and which not

File: lib/secure-boot-lib.robot (hunk 2 above)
Comment (pietrushnic) [reply lib/secure-boot-lib.robot:239]: @macpijan, please let me know next time that it doesn't make sense to spend time when you plan to solve stuff yourself.

File: lib/secure-boot-lib.robot (hunk 4 above)
Comment (pietrushnic) [reply lib/secure-boot-lib.robot:269]: > That was in my MR (submitted before this one), which means you have probably not looked at that before submitting this. 

@macpijan it is very hard to follow your process. I'm sorry, but I'm not able to sync in 2-3 hours with everything you publish. Changes come in big chunks and make previous assumptions obsolete. It is very hard to make meaningful contributions working that way. It is definitely my fault that I didn't notice the comment you pointed to. I essentially spent most of my contribution time trying to understand the changes you introduced, and the approach was so different from the previous set of keywords that I was lost. Rebasing branches on which someone else works is damaging.

**Comment by pietrushnic** [comment]
Dropping https://github.com/Dasharo/open-source-firmware-validation/pull/98 in favor of https://github.com/Dasharo/open-source-firmware-validation/pull/95
